services/products.js

```diff
@@ -0,0 +1,141 @@
+const express = require('express');
+const router = express.Router();
+
+const {
+  getAllProducts,
+  calculateQuantity,
+  returnCategory,
+  returnUltraSale,
+  returnFeatureProducts,
+  sortBy,
+} = require('../core/productsHelper');
+
+//homepage
+router.get('/', async (req, res, next) => {
+  const categoriesNumber = await calculateQuantity();
+  const products = await getAllProducts();
+  const ultraSaleProducts = await returnUltraSale();
+  const featureProducts = await returnFeatureProducts();
+
+  res.render('main', {
+    homePage: true,
+    categoriesNumber,
+    products,
+    ultraSaleProducts,
+    featureProducts,
+  });
+});
+
+//product list page
+router.get('/products/', async (req, res, next) => {
+  const category = req.query.category;
+
+  let categoryName = '';
+  let productsTotal = null;
+  let outputProducts = null;
+
+  if (category === 'all') {
+    categoryName = 'All products';
+    productsTotal = await getAllProducts();
+  } else {
+    switch (category) {
+      case 'meats':
+        categoryName = 'Fresh meats';
```

Comment:
pls add categoryName to the database. There values should be dynamic and not hardcoded here